### PR TITLE
New rule S4158: Empty collections should not be accessed or iterated - for VB.NET

### LINKED ISSUE #7303: New rule S4158: Empty collections should not be accessed or iterated - for VB.NET
Implement S4158 for VB.NET

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Fixes #7303

I removed stuff that was redundant with C# and cover only the main topics. 

RSPEC PR: https://github.com/SonarSource/rspec/pull/2181

### CODE REVIEW

File: analyzers/its/expected/Ember-MM/scraper.EmberCore-{EF6A550E-DD76-4F4D-8250-8598140F828B}-S3900.json [hunk 1]
```diff
@@ -58,19 +58,6 @@
 "location":  {
 "uri":  "sources\Ember-MM\Addons\scraper.EmberCore\Scraper\clsScrapeImages.vb",
 "region":  {
-"startLine":  435,
```
Comment (pavel-mikula-sonarsource) [inline analyzers/its/expected/Ember-MM/scraper.EmberCore-{EF6A550E-DD76-4F4D-8250-8598140F828B}-S3900.json:61]: This is extremly long method -> probably lost issue due to size limitations

File: analyzers/its/expected/Ember-MM/generic.EmberCore.NMT-{84B2143A-D04F-4262-923D-21AEDF86E2B7}-S2259.json [hunk 2]
```diff
@@ -158,19 +158,6 @@
 },
 {
 "id":  "S2259",
-"message":  "'c.values.FirstOrDefault(Function(y) y.label = saved)' is Nothing on at least one execution path.",
-"location":  {
-"uri":  "sources\Ember-MM\Addons\generic.EmberCore.NMT\dlgNMTMovies.vb",
-"region":  {
-"startLine":  960,
```
Comment (pavel-mikula-sonarsource) [inline analyzers/its/expected/Ember-MM/generic.EmberCore.NMT-{84B2143A-D04F-4262-923D-21AEDF86E2B7}-S2259.json:165]: Lost TP. Not sure why, the method is not so big

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.vb [hunk 3]
```diff
@@ -0,0 +1,333 @@
+﻿Imports System
+Imports System.Collections.Generic
```
Comment (mary-georgiou-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.vb:2]: Those two are not needed.

File: analyzers/its/expected/Ember-MM/generic.EmberCore.NMT-{84B2143A-D04F-4262-923D-21AEDF86E2B7}-S2259.json (hunk 2 above)
Comment (mary-georgiou-sonarsource) [reply analyzers/its/expected/Ember-MM/generic.EmberCore.NMT-{84B2143A-D04F-4262-923D-21AEDF86E2B7}-S2259.json:165]: Not clear to me why this is a TP :/

**Review by mary-georgiou-sonarsource — APPROVED** [review]
LGTM - Not 100% confident about VB UTs
I checked that you cover most of the cases from C# but not sure if something could be done better or you are missing some particular VB syntax.

File: analyzers/its/expected/Ember-MM/generic.EmberCore.NMT-{84B2143A-D04F-4262-923D-21AEDF86E2B7}-S2259.json (hunk 2 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/its/expected/Ember-MM/generic.EmberCore.NMT-{84B2143A-D04F-4262-923D-21AEDF86E2B7}-S2259.json:165]: `FirstOrDefault` returns something or `Nothing`. On the `Nothing` case, it's TP

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.vb (hunk 3 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.vb:2]: Will fix in next PR

**Pull request merged**